packages/ui/src/components/organisms/chart/chart.utils.ts

```diff
@@ -1,32 +1,61 @@
+import simplify from "simplify-js";
 import { ChartInterval, ChartTickInterval } from "./chart.types";
 
 type DateParams = number | string | Date;
+type Point = { x: number; y: number };
 
 /**
  * Downsamples data to a maximum number of data points.
  * Useful for displaying large datasets in charts to avoid performance issues.
  */
-export const getDownsampledData = <T>(
+
+export const getSummarizedData = <T extends Point>(
   data: T[],
   options: {
     /**
      * Threshold for the maximum number of data points to display.
      */
     maximumDataPoints: number;
   } = { maximumDataPoints: 100 },
-) => {
-  if (data.length > options.maximumDataPoints) {
-    const downsampled: T[] = [];
-    const interval = Math.ceil(data.length / options.maximumDataPoints);
+): {
+  downsampled: Point[];
+  domain: number[];
+  max: number;
+  min: number;
+  range: number;
+  average: number;
+} => {
+  if (!data.length) {
+    return {
+      downsampled: [],
+      domain: [],
+      max: 0,
+      min: 0,
+      range: 0,
+      average: 0,
+    };
+  }
 
-    for (let i = 0; i < data.length; i += interval) {
-      downsampled.push(data[i] as T);
+  const domain = data.map((it) => it.y ?? 0);
+  const max = Math.max(...domain);
+  const min = Math.min(...domain);
+  const range = max - min;
+  // The `simplify` function requires a threshold value to determine the tolerance for simplification.
+  // Because each chart has a different range, this calculation may not be perfect and more cases may need to be handled.
+  const threshold = (() => {
```

Comment:
Is there a meaningful difference between the term threshold and tolerance? I'm having trouble grokking what this does.